src/main/java/com/google/sps/model/ActionableMessageHelperImpl.java

```diff
@@ -0,0 +1,117 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.model;
+
+import com.google.api.services.gmail.model.Message;
+import com.google.api.services.gmail.model.MessagePartHeader;
+import com.google.sps.exceptions.GmailMessageFormatException;
+import com.google.sps.utility.GmailUtility;
+import java.util.Arrays;
+import java.util.List;
+
+/** Contains methods that help create the non-trivial fields in an ActionableMessage. */
+public final class ActionableMessageHelperImpl implements ActionableMessageHelper {
+  // Public for testing purposes
+  public static final String UNDISCLOSED_RECIPIENTS_TO_HEADER = "undisclosed-recipients:;";
+  public static final int MAILING_LIST_LOWER_BOUND = 15;
```

Comment:
Did you mean the class is public for testing purposes?

Reply:
No, the properties themselves are public. These are the same values that should be used in the tests as well